File: Java/StackArray.java
public class StackArray {

	int [] stack;
	int top = -1;

	public void push(int val){
		top++;
		stack[top] = val;
	}

	public int peek(){
		return stack[top];
	}

	public void pop(){
		top--;
	}

	public boolean isEmpty(){
		if (top != -1){
			return true;
		}
		else {
			return false;
		}
	}

	public void printList(){
		for (int x: stack){
			System.out.println(x);
		}
	}

	public static void main(String [] args){
		StackArray stackArray = new StackArray();
		System.out.println(stackArray.isEmpty());

		stackArray.push(1);
		stackArray.push(3);
		stackArray.push(5);
		stackArray.push(7);
		stackArray.push(9);

		stackArray.printList();
		System.out.println(stackArray.isEmpty());	
		System.out.println(stackArray.peek());
		stackArray.pop();

		System.out.println("Hello World!");
	}
}

File: C/StackArray.c
#include <stdio.h>
#include <stdbool.h>

int stack[10];
int top = -1;

void pop();
int peek();
void push(int data);
bool isEmpty();
void printList();

int main(){
	printf("%d\n", isEmpty());
	
	push(1);
	push(3);
	push(5);
	push(7);
	push(9);

	printf("%d\n", peek());
	printList();
	pop();
	printf("%d\n", isEmpty());

	return 0;
}

void pop(){
	top--;
}

int peek(){
	return stack[top];
}

void push(int data){
	top++;
	stack[top] = data;
}

bool isEmpty(){
	if (top != -1){
		return true;
	}
	else {
		return false;
	}
}

void printList(){
	for(int i = 0; i < 10; i++){
		printf("%d\n", stack[i]);
	}
}
